### ISSUE TITLE
Enabling Location with mode High Accuracy or Battery saving without user needing to visit Settings

### ISSUE THREAD

**Issue by rohannexialabs** [mrmans0n/smart-location-lib#102]
First of all, brilliant library, thanks for making it public. Now, to the problem, as question states, how to enable Location Service without sending user to the Settings screen. I have seen some apps do that including Tinder and Google Maps. Related stackoverflow question can be found @ http://stackoverflow.com/q/28759454/1979347.

**Comment by PsyGik** [comment]
It's easy. The answer to the SO post is the correct way.

Some snippets,

    /**
     * Prompt user to enable GPS and Location Services
     * @param mGoogleApiClient
     * @param activity
     */
    public static void locationChecker(GoogleApiClient mGoogleApiClient, final Activity activity) {
        LocationRequest locationRequest = LocationRequest.create();
        locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
        locationRequest.setInterval(30 * 1000);
        locationRequest.setFastestInterval(5 * 1000);
        LocationSettingsRequest.Builder builder = new LocationSettingsRequest.Builder()
                .addLocationRequest(locationRequest);
        builder.setAlwaysShow(true);
        PendingResult<LocationSettingsResult> result =
                LocationServices.SettingsApi.checkLocationSettings(mGoogleApiClient, builder.build());
        result.setResultCallback(new ResultCallback<LocationSettingsResult>() {
            @Override
            public void onResult(LocationSettingsResult result) {
                final Status status = result.getStatus();
                final LocationSettingsStates state = result.getLocationSettingsStates();
                switch (status.getStatusCode()) {
                    case LocationSettingsStatusCodes.SUCCESS:
                        // All location settings are satisfied. The client can initialize location
                        // requests here.
                        break;
                    case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
                        // Location settings are not satisfied. But could be fixed by showing the user
                        // a dialog.
                        try {
                            // Show the dialog by calling startResolutionForResult(),
                            // and check the result in onActivityResult().
                            status.startResolutionForResult(
                                    activity, 1000);
                        } catch (IntentSender.SendIntentException e) {
                            // Ignore the error.
                        }
                        break;
                    case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
                        // Location settings are not satisfied. However, we have no way to fix the
                        // settings so we won't show the dialog.
                        break;
                }
            }
        });
    }

And use it like,

    mGoogleApiClient = new GoogleApiClient
                .Builder(this)
                .enableAutoManage(this, 34992, this)
                .addApi(LocationServices.API)
                .addConnectionCallbacks(this)
                .addOnConnectionFailedListener(this)
                .build();

        locationChecker(mGoogleApiClient, MyActivity.this);

Basically your user will get a prompt like this, where they can enable Location in High Accuracy mode without going in the settings
![Turn on GPS](http://i.stack.imgur.com/81TfP.png)

**Comment by rohannexialabs** [comment]
@PsyGik Thanks for the detailed tutorial. I know how to get this when I am using `GoogleApiClient ` API, but since I am using this library, it doesn't make sense to create these objects again. Since, the library is already using them internally (I think), I wanted to ask is there a more simple (memory effective) way. I think I should have made myself more clearer.

**Comment by PsyGik** [comment]
@rohannexialabs, all you had to do was search.
https://github.com/mrmans0n/smart-location-lib/blob/0ce6dfee098c5105e857a90d724aa22fb1611f96/library/src/main/java/io/nlopez/smartlocation/location/providers/LocationGooglePlayServicesProvider.java

**Comment by rohannexialabs** [comment]
@PsyGik I should have been more thorough, sorry for that and thanks for the help.

**Comment by kalpeshsantoki07** [comment]
@rohannexialabs Can you let me know how request for enable dialog with this library ?

**Comment by JobGetabu** [comment]
/**
     * Sets whether or not we should request (before starting updates) the availability of the
     * location settings and act upon it.
     *
     * @param allowingLocationSettings TRUE to show the dialog if needed, FALSE otherwise (default)
     */
    public void setCheckLocationSettings(boolean allowingLocationSettings) {
        this.checkLocationSettings = allowingLocationSettings;
    }

Still does not prompt dialogue when set to true
